### PR TITLE
Add option to allow hidden field validation #16

### PR DESCRIPTION
**Pull request by dotnetshadow**
Signed-off-by: DotnetShadow <296247+dotnetshadow@users.noreply.github.com>

### CODE REVIEW

File: README.MD [hunk 1]
```diff
@@ -218,4 +218,10 @@ You can also supply a callback function to be run after the check.
 
 ```ts
 v.isFieldValid(document.getElementById('field'), true, myCallbackFn)
+```
+
+## Hidden fields validation
+By default validation is skipped for hidden fieldss. To enable validation for hidden fields validation use:
```
Comment (haacked) [inline README.MD:224]:
```suggestion
By default validation is skipped for hidden fields. To enable validation for hidden fields validation use:
```

**Review by haacked — CHANGES_REQUESTED** [review]

File: src/index.ts [hunk 2]
```diff
@@ -924,7 +929,7 @@ export class ValidationService {
      * @returns
      */
     private isHidden(input: HTMLElement) {
-        return !( input.offsetWidth || input.offsetHeight || input.getClientRects().length );
+        return !(this.allowHiddenFields || input.offsetWidth || input.offsetHeight || input.getClientRects().length );
```
Comment (haacked) [inline src/index.ts:932]: Does this mean by default, we don’t validate fields that are hidden because they’ve scrolled offscreen?

Comment (haacked) [reply src/index.ts:932]: Ah, looks like `offsetWidth` has nothing to do if it’s currently visible in the viewport.

Comment (dotnetshadow) [reply src/index.ts:932]: Yeah, I left the original logic, I'm guessing it knows how to figure it out

**Comment by dotnetshadow** [comment]
Fixed the typo, thanks for catching it

**Comment by dotnetshadow** [comment]
Do I have to do anything else?

**Comment by haacked** [comment]
This fixes #16

**Pull request merged**